Makefile

```diff
@@ -62,64 +65,72 @@ BUILD_IMAGE ?= golang:1.9-alpine
 all: build
 
 build-%:
-	@$(MAKE) --no-print-directory ARCH=$* build
+	@$(MAKE) --no-print-directory ARCH=$(word 2,$(subst -, ,$*)) OS=$(word 1,$(subst -, ,$*)) build
 
 container-%:
-	@$(MAKE) --no-print-directory ARCH=$* container
+	@$(MAKE) --no-print-directory ARCH=$(word 2,$(subst -, ,$*)) OS=$(word 1,$(subst -, ,$*)) container
 
 push-%:
-	@$(MAKE) --no-print-directory ARCH=$* push
+	@$(MAKE) --no-print-directory ARCH=$(word 2,$(subst -, ,$*)) OS=$(word 1,$(subst -, ,$*)) push
 
-all-build: $(addprefix build-, $(ALL_ARCH))
+all-build: $(addprefix build-, $(BUILD_PLATFORMS))
 
-all-container: $(addprefix container-, $(ALL_ARCH))
+all-container: $(addprefix container-, $(CONTAINER_PLATFORMS))
 
-all-push: $(addprefix push-, $(ALL_ARCH))
+all-push: $(addprefix push-, $(CONTAINER_PLATFORMS))
 
-build: bin/$(ARCH)/$(BIN)
+build: bin/$(OS)-$(ARCH)/$(BIN)
 
-bin/$(ARCH)/$(BIN): build-dirs
+bin/$(OS)-$(ARCH)/$(BIN): build-dirs
+ifeq ($(filter $(OS)-$(ARCH),$(BUILD_PLATFORMS)),)
```

Comment:
This is the thing that would prohibit "out of the box", support for an unlisted platform to just get a build running.

Reply:
presume that people will edit BUILD_PLATFORMS prior to `make all-build` to suit their desired platforms.

Reply:
👍